pkg/internal/testutils/generate_pod_yamls.go

```diff
@@ -0,0 +1,204 @@
+	return filepath.Walk(origDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() { // nothing to do
+			return nil
+		}
+		origFile := filepath.Join(origDir, info.Name())
+		tempFile := filepath.Join(tempDir, info.Name())
+		// if the file contains workload with given srcName or dstName : get the workload object and
+		// generate matching pod yaml file in the temp dir
+		// this is needed since we need also a copy of the workload labels which may be used in the netpols rules
+
+		// find src and get its labels
+		srcLabels, err := checkFileContainsInputWorkloadAndGetItsLabels(origFile, srcName, srcNs)
+		if err != nil {
+			return err
+		}
+		if srcLabels != nil { // nil means the src was not found in that file
+			err = generatePodYaml(tempDir, srcName, srcNs, srcLabels)
+			if err != nil {
+				return err
+			}
+		}
+
+		// find dst and get its labels
+		dstLabels, err := checkFileContainsInputWorkloadAndGetItsLabels(origFile, dstName, dstNs)
+		if err != nil {
+			return err
+		}
+		if dstLabels != nil { // nil means the dst was not found in that file
+			err = generatePodYaml(tempDir, dstName, dstNs, dstLabels)
+			if err != nil {
+				return err
+			}
+		}
+		// copy the orig file (having objects with kinds other than ns, pod, netpols will not affect the result
+		// since it will not be parsed with the eval command)
+		return copyFile(origFile, tempFile)
+	})
+}
+
+// PodInfo contains metadata of the pod so we can :
+// 1. extract relevant workload from input resources
+// 2. generate relevant pod template
+type PodInfo struct {
+	Name      string            `yaml:"name"`
+	Namespace string            `yaml:"namespace,omitempty"`
+	Labels    map[string]string `yaml:"labels,omitempty"`
+}
+
+// workloadMetadata the yaml is unmarshal to workload metadata struct which is the only interesting part for our goals
+type WorkloadMetadata struct {
+	Metadata PodInfo `yaml:"metadata"`
+}
+
+// checkFileContainsInputWorkloadAndGetItsLabels gets yaml contents and checks if it contains a workload object with the
+// given name and namespace
+// note that this assumes that if the object name and namespace matches the input, it is the workload object
+// since our tests-dirs contain unique names for workloads (different than policies names)
+// @todo : verify the kind is of a workload type too
+func checkFileContainsInputWorkloadAndGetItsLabels(origFile, podName, podNs string) (map[string]string, error) {
+	fileContents, err := os.ReadFile(origFile)
+	if err != nil {
+		return nil, err
+	}
+	// Splitting the YAML into multiple documents
+	docs := splitYamlDocs(fileContents)
+
+	// we are interested in Metadata of the workload only.
+	// Iterate through objects to find the matching one
+	for _, doc := range docs {
+		var obj WorkloadMetadata
+		if err := yaml.Unmarshal(doc, &obj); err != nil {
+			return nil, err
+		}
+		if obj.Metadata.Name == podName && (obj.Metadata.Namespace == podNs ||
```

Comment:
can use instead the already existing functions from `manifests.GetResourceInfosFromDirPath()` and 
 `parser.ResourceInfoListToK8sObjectsList()` ?